### DISCUSSION TITLE
Add an `enableRichTextLink` on collection configs

### DISCUSSION THREAD

**Discussion by jakehopking** [payloadcms/payload#2567] [Feature Requests & Ideas]
Can we add a simple feature to collection configs that allows us to control which collections populate the RichText link autocomplete dropdown — which would work in a similar way to how the `enableRichTextRelationship` controls which collections populate the RichText relationships dialog.

For example I have 'customer', 'order' and 'user' collections that I do not want to appear inside the 'Internal Link' list.

**Comment by DanRibbens** [comment]
Thanks for the submitting @jakehopking!

Once this is merged, https://github.com/payloadcms/payload/pull/2559 you'll have this capability in a much less direct way than your proposal. It could be a good workaround in the meantime.

Instead of having a `enableRichTextLink` as it's own property, with the #2559 you open access to the fields used for the RichText links. This would require you to find the relationship field in the `defaultFields` and replace the `relationTo` options as needed.

**Reply by jakehopking** [reply]
Does this mean we’ll need to define these relationships on each instance of a RichText field?
At this stage I don’t understand how to do what you’ve mentioned — will there be an example of how to do this in the docs?

FWIW: I feel that defining which collections should populate the links would make sense to sit alongside the `enableRichTextRelationship` property.

**Reply by DanRibbens** [reply]
Well 2559 hasn't been released yet. I was giving workarounds because I don't know when we'll get to adding `enableRichTextLink` as a feature.

For anyone in the community it would be a fairly small PR to make this happen:

1) add `enableRichTextLink?: boolean` to the richtext field admin type
2) add to the admin richtext schema (joi validation)
3) filter the `doc` relationTo in src/admin/components/forms/field-types/RichText/elements/link/LinkDrawer/baseFields.ts using the new property
4) add e2e test
5) documentation

**Comment by DanRibbens** [comment]
Looks like @mvarendorff has a jump on this https://github.com/payloadcms/payload/pull/2560

**Comment by DanRibbens** [comment]
Hey @jakehopking,
Collection `admin.enableRichTextLink` is now available in Payload v1.7.5.

**Reply by jakehopking** [reply]
This is super! Thanks! 🥳
